skycoin/skywire
# Feature/remove redialing

Did you run `make format && make check`?

Fixes #830
Fixes #844
Fixes #843

 Changes:	
- Remove transport redialing
- Remove transport statuses as a concept
- Add persistent transports configuration (transports that are repeatedly redialed when broken, and are set up at visor startup)

How to test this PR:
- Run a couple of visors and establish transports between them manually, delete those transports manually, crash visors deliberately.
- Use transport setup to add skywire transports and public visors and check that the connections are properly get set up and are not reconnected.
- Add persistent transports to configuration and check that it gets established on startup and reestablished when the target visor goes offline and back

A known issue is that sudph transports do not get closed automatically, when other end of connection closes. This could be an issue with kcp-go library configuration, or it could be the case that the library doesn't support it.
For this case, [I added an automatic connection close logic](https://github.com/skycoin/skywire/pull/842/files#diff-9a822d96295268985a33b3260cfe31fae865ab929e735feb6e2427e2b1687a57R330), built on `net.Conn` deadline logic. Basically, the idea is to update deadline every time the connection is used.
For this to work properly, we need to add support for some kind of keep-alive messages to the transport layer. 
Currently, it is disabled (deadline is 0 and ignored by conn). 

Shutdown logic is also fixed in this PR, since I thought it would help with sudph issues.
